### ISSUE TITLE
astUtils.isDecimalInteger should return `true` for NonOctalDecimalIntegerLiteral literals

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13588]
**Tell us about your environment**

* **ESLint Version:** 7.7.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2015
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogZXJyb3IgKi9cblxuKDA4KS50b1N0cmluZygpO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

```js
/* eslint no-extra-parens: error */

(08).toString();
```

```bash
npx eslint index.js --fix
```

**What did you expect to happen?**

No error, because the rule doesn't report errors on other number literals that can be continued with decimal dot.

**What actually happened? Please include the actual, raw output from ESLint.**

Error and autofix to:

```js
/* eslint no-extra-parens: error */

08.toString();
```

Error on the fixed code:

```
  3:4  error  Parsing error: Identifier directly after number
```

**Are you willing to submit a pull request to fix this bug?**

Yes, after we merge https://github.com/eslint/eslint/pull/13581
